[GHF] Add ability to specify team as the approver

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #74887
* #74893
* __->__ #74892

All GitHub teams belong to organization, and therefore must be
referenced as `org/team_name`

Add memoizable `gh_get_team_members` and tests that it can be used to
query existing and non-existing teams

Note: Requires "org.read" permission on the token to work function

malfet committed:
[GHF] Add ability to specify team as the approver

All GitHub teams belong to organization, and therefore must be
referenced as `org/team_name`

Add memoizable `gh_get_team_members` and tests that it can be used to
query existing and non-existing teams

Note: Requires "org.read" permission on the token to work function

[ghstack-poisoned]

malfet committed:
Update on "[GHF] Add ability to specify team as the approver"


All GitHub teams belong to organization, and therefore must be
referenced as `org/team_name`

Add memoizable `gh_get_team_members` and tests that it can be used to
query existing and non-existing teams

Note: Requires "org.read" permission on the token to work function

[ghstack-poisoned]

janeyx99 commented on .github/scripts/trymerge.py:
```diff
@@ -591,7 +627,13 @@ def find_matching_merge_rule(pr: GitHubPR, repo: GitRepo, force: bool = False) -
-        rule_approvers_set = set(rule.approved_by)
+        rule_approvers_set = set()
+        for approver in rule.approved_by:
+            if "/" in approver:
```
are all teams org/name format?

seemethere replied:
They should be

malfet replied:
Yes they are, according to https://github.blog/2012-05-09-introducing-team-mentions/

malfet committed:
Update on "[GHF] Add ability to specify team as the approver"


All GitHub teams belong to organization, and therefore must be
referenced as `org/team_name`

Add memoizable `gh_get_team_members` and tests that it can be used to
query existing and non-existing teams

Note: Requires "org.read" permission on the token to work function

[ghstack-poisoned]
